File: nysol_python/mcmd/methods/convTex/db/smp/mtraflg.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='mtraflg'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
customer,egg,milk
A,1,1
B,,1
C,1,
D,1,1
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
``egg`` と ``milk`` 項目の値がNULL値以外なら、それら項目名を要素としたベクトルを作成する。
'''
script['sh_code']='''
mtraflg f=egg,milk a=transaction i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mtraflg(f="egg,milk", a="transaction", i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

script={}
script['title']='基本例2'
script['comment']='''
出力された結果を ``r=True`` をつけて再実行し元に戻す。
'''
script['sh_code']='''
mtraflg -r f=egg,milk a=transaction i=rsl1.csv o=rsl2.csv
'''
script['py_code']='''
nm.mtraflg(r=True, f="egg,milk", a="transaction", i="rsl1.csv", o="rsl2.csv").run()
'''
script['odatas']='rsl2.csv'
db['scripts'].append(script)

script={}
script['title']='区切り文字の指定'
script['comment']='''
区切り文字を-(ハイフン）で連結し、 ``transaction`` という項目名で出力する。
'''
script['sh_code']='''
mtraflg f=egg,milk a=transaction delim=- i=dat1.csv o=rsl3.csv
'''
script['py_code']='''
nm.mtraflg(f="egg,milk", a="transaction", delim="-", i="dat1.csv", o="rsl3.csv").run()
'''
script['odatas']='rsl3.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/mtraflg.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
customer,egg,milk
A,1,1
B,,1
C,1,
D,1,1
EOF
)}

############## Example 1
title="Basic Example"
comment=<<'EOF'
Create a string of vector from the list of non-null values in column \verb|egg| and \verb|milk|.
EOF
scp=<<'EOF'
more dat1.csv
mtraflg f=egg,milk a=transaction i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

############## Example 2
title="Basic Example 2"
comment=<<'EOF'
Use \verb|-r| option to revert the output results back to the original data.
EOF
scp=<<'EOF'
mtraflg -r f=egg,milk a=transaction i=rsl1.csv o=rsl2.csv
more rsl2.csv
EOF
run(scp,title,comment)

############## Example 3
title="Specify the delimiter"
comment=<<'EOF'
Combine items using the “-” (hyphen) as delimiter. Save output in column named \verb|transaction|.
EOF
scp=<<'EOF'
mtraflg f=egg,milk a=transaction delim=- i=dat1.csv o=rsl3.csv
more rsl3.csv
EOF
run(scp,title,comment)
